Userland/Libraries/LibGL/SoftwareRasterizer.cpp

```diff
@@ -382,6 +382,7 @@ static void rasterize_triangle(const RasterizerOptions& options, Gfx::Bitmap& re
                             + FloatVector4(float_dst.w(), float_dst.w(), float_dst.w(), float_dst.w()) * dst_factor_dst_alpha;
 
                         *dst = to_rgba32(*src * src_factor + float_dst * dst_factor);
+                        *dst &= options.color_mask;
```

Comment:
This is not correct. We don't want to completely black out color components in the render target. instead we want to mask out the write itself, so the original content of the color channel remains unmodified in the render target.